cclib/io/ccio.py

```diff
@@ -161,24 +141,35 @@ def ccread(source, *args, **kwargs):
     Returns:
         a ccData object containing cclib data attributes
     """
-
-    log = ccopen(source, *args, **kwargs)
-    if log:
-        if kwargs.get("verbose", None):
-            print(f"Identified logfile to be in {log.logname} format")
-        # If the input file is a CJSON file and not a standard compchemlog file
-        cjson_as_input = kwargs.get("cjson", False)
-        if cjson_as_input:
-            return log.read_cjson()
+    log = None
+    try:
+        log = ccopen(source, *args, **kwargs)
+        if log:
+            if kwargs.get("verbose", None):
+                print(f"Identified logfile to be in {log.logname} format")
+            # If the input file is a CJSON file and not a standard compchemlog file
+            cjson_as_input = kwargs.get("cjson", False)
+            if cjson_as_input:
+                return log.read_cjson()
+            else:
+                return log.parse()
         else:
-            return log.parse()
-    else:
-        if kwargs.get('verbose', None):
-            print('Attempting to use fallback mechanism to read file')
-        return fallback(source)
-
-
-def ccopen(source, *args, **kwargs):
+            if kwargs.get('verbose', None):
```

Comment:
Not originally your code, but this should be
```suggestion
            if kwargs.get('verbose', False):
```

Reply:
Thanks good spot. I'll change this to be just a normal kwarg because I don't see why we'd need to use a kwarg dict here.

Reply:
Ok it's because the kwargs dict is passed to `ccread`, which passes it unmodified to the `LogFile` or `filereader.Reader` constructor, neither of which inspect for the `verbose` key 🙃. I'll replace all these weird kwarg dicts with proper signatures.